src/index.ts

```diff
@@ -56,6 +57,45 @@ passport.use('local', new LocalStrategy(
   }
 ))
 
+if ( process.env.GOOGLE_CLIENT_ID && process.env.GOOGLE_CLIENT_SECRET ) {
+  passport.use('google', new GoogleStrategy({
+      clientID: process.env.GOOGLE_CLIENT_ID,
+      clientSecret: process.env.GOOGLE_CLIENT_SECRET,
+      callbackURL: '/auth/google/redirect',
+    },
+    (issuer, profile, done  ) => {
+      var username = profile.emails[0].value.trim()
+      db.get(username)
+        .then((doc: any) => {
+          return done(null, doc)
+        })
+        .catch(err => {
+          if (err.message === 'missing' && config.addSSOUsers) {
+            db.put({
+              _id: username,
+              admin: false,
+              pfp: '/img/default-pfps/1.png',
+              wishlist: []
+            })
+            db.get(username)
+              .then((doc: any) => {
+                return done(null, doc)
+              })
+              .catch(err => {
+                console.log(err)
+                return done(null, false, { message: err })
+              })          
+          }
+          else {
+            if (err.message === 'missing') return done(null, false, { message: 'Unknown user.' })
```

Comment:
Ideally, this should use translations. I see that the local strategy doesn't use them either and it may not be possible in the index.ts file with my current architecture. If it can't be done without major changes, I'm okay with it.

Reply:
Localization and English translations added with https://github.com/Wingysam/Christmas-Community/pull/138/commits/736e537ee1b3c2b18bbf78b184750d4577707ac4